paddle/phi/kernels/gpu/multi_tensor_adam_kernel.cu

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/phi/kernels/multi_tensor_adam_kernel.h"
+#include <assert.h>
+#include <cstdlib>
+#include <vector>
+#include "paddle/fluid/framework/tensor_util.h"
+#include "paddle/phi/backends/gpu/gpu_context.h"
+#include "paddle/phi/common/amp_type_traits.h"
+#include "paddle/phi/common/float16.h"
+#include "paddle/phi/common/scalar.h"
+#include "paddle/phi/core/dense_tensor.h"
+#include "paddle/phi/core/kernel_registry.h"
+#include "paddle/phi/core/tensor_utils.h"
+#include "paddle/phi/kernels/funcs/adam_functors.h"
+#include "paddle/phi/kernels/funcs/for_range.h"
+#include "paddle/phi/kernels/funcs/selected_rows_functor.h"
+#include "paddle/phi/kernels/multi_tensor_adam_utility_kernel.cuh"
+
+namespace phi {
+
+#define BLOCK_SIZE 512
+#define PARALLEL_SIZE 4
+
+template <typename T, typename MT, int N>
+struct CudaKernelFunctor {
+  __device__ __forceinline__ void operator()(int compute_group_size,
+                                             TensorAndBlockInf<N> tabi,
```

Comment:
The name `tabi` is quite hard to understand. I know it is short for `TensorAndBlockInf`, but it is not quite obvious to understand.